rllib/algorithms/ppo/tests/test_ppo_with_rl_module.py

```diff
@@ -1,18 +1,93 @@
-import numpy as np
 import unittest
 
+import numpy as np
+import tree
+import gym
+import torch
+
 import ray
 import ray.rllib.algorithms.ppo as ppo
-
 from ray.rllib.algorithms.callbacks import DefaultCallbacks
+from ray.rllib.algorithms.ppo.torch.ppo_torch_rl_module import (
+    PPOModuleConfig,
+    PPOTorchRLModule,
+    get_ppo_loss,
+)
+from ray.rllib.core.rl_module.encoder import (
+    IdentityConfig,
+    FCConfig,
+    LSTMConfig,
+)
 from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID
+from ray.rllib.policy.sample_batch import SampleBatch
 from ray.rllib.utils.metrics.learner_info import LEARNER_INFO, LEARNER_STATS_KEY
+from ray.rllib.utils.numpy import convert_to_numpy
 from ray.rllib.utils.test_utils import (
     check,
     check_compute_single_action,
     check_train_results,
     framework_iterator,
 )
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+
+
+# Get a model config that we expect from the catalog
+def get_expected_model_config(env, lstm, shared_encoder):
+    assert not len(env.observation_space.shape) == 3, "Implement VisionNet first!"
+    obs_dim = env.observation_space.shape[0]
+
+    if shared_encoder:
+        assert not lstm, "LSTM can only be used in PI"
+        shared_encoder_config = FCConfig(
+            input_dim=obs_dim,
+            hidden_layers=[32],
+            activation="ReLU",
+            output_dim=32,
+        )
+        pi_encoder_config = IdentityConfig(output_dim=32)
+        vf_encoder_config = IdentityConfig(output_dim=32)
+    else:
+        shared_encoder_config = IdentityConfig(output_dim=obs_dim)
+        if lstm:
+            pi_encoder_config = LSTMConfig(
+                input_dim=obs_dim,
+                hidden_dim=32,
+                batch_first=True,
+                output_dim=32,
+                num_layers=1,
+            )
+        else:
+            pi_encoder_config = FCConfig(
+                input_dim=obs_dim,
+                output_dim=32,
+                hidden_layers=[32],
+                activation="ReLU",
+            )
+        vf_encoder_config = FCConfig(
+            input_dim=obs_dim,
+            output_dim=32,
+            hidden_layers=[32],
+            activation="ReLU",
+        )
+
+    pi_config = FCConfig()
+    vf_config = FCConfig()
```

Comment:
These are technically all FC encoder now (shared encoder, pi encoder, vf encoder, pi and vf). In a follow up PR the first encoders should be able to be convolutional though.